### ISSUE TITLE
[Netflow] Support TSDS

### ISSUE THREAD

**Issue by BenB196** [elastic/integrations#7549]
Hi All,

I was curious if there are any plans to support enabling TSDS on the Netflow integration.

While this integration currently falls under the `logs` type. I think there would be significant value in allowing this integration to leverage TSDS.

Netflow contains a large number of metrics and generally at scale, will generate a significant number of timeseries events that need to be indexed and stored. I think that the Netflow integration would receive a significant value increase by leveraging TSDS in indexing speed, storage usage, and search/agg speed.

**Comment by andrewkroh** [comment]
I was having the same idea, but for aws.vpcflow data which is very similar just with a smaller number of possible fields. I think we should do a test with using TSDS on one of these flow log data sources. I think storage size would be the biggest benefit.

One thing that could cause an issue (particularly in the aws.vpcflow case) is late arriving data (like if it is historical data read from S3). TSDS can only accept data that has a "recent" timestamp (see https://www.elastic.co/guide/en/elasticsearch/reference/current/tsds.html#tsds-accepted-time-range).

**Comment by BenB196** [comment]
Interesting, I didn't really think about historical ("backfill") data here, but it does make sense as to consider.

I wonder if https://github.com/elastic/elasticsearch/issues/98463 would help as well in this scenario. I don't really know the dynamics around use cases like importing VPC flow data, so not sure how "useable" this feature would be in a "backfill" scenario.

**Comment by jamiehynds** [comment]
Related presentation from ElatiFlow: https://www.youtube.com/watch?v=MuMXNTFsKto&t=1238s&ab_channel=OfficialElasticCommunity

**Comment by BenB196** [comment]
I think one of the main challenges with TSDS going back to something @andrewkroh pointed out and that is the possibility of needing to backfill data, which doesn't have the greatest experience with TSDS.

Elastic is working on a new "LogsDB" index mode, https://github.com/elastic/elasticsearch/issues/106462, which _hopefully_ will provide many of the same benefits of TSDS, without the challenges of backfilling data. I don't think it will be 100% as efficient as TSDS could possibly be, but would still be a nice value add.

**Comment by pkoutsovasilis** [comment]
> I think one of the main challenges with TSDS going back to something @andrewkroh pointed out and that is the possibility of needing to backfill data, which doesn't have the greatest experience with TSDS.
> 
> Elastic is working on a new "LogsDB" index mode, [elastic/elasticsearch#106462](https://github.com/elastic/elasticsearch/issues/106462), which _hopefully_ will provide many of the same benefits of TSDS, without the challenges of backfilling data. I don't think it will be 100% as efficient as TSDS could possibly be, but would still be a nice value add.

yes! `LogsDB` index mode sounds like the best of both worlds with an acceptable trade-off. This comes as `Tech Preview` in ES 8.14.0 but I think the target is reach `GA` in 8.15.0

**Comment by andrewkroh** [comment]
LogsDB should already be usable (as it is with most log data streams). You can test it out by adding `index.mode: logsdb` into the index template[^1]. Every integration supports a `@custom` component template for adding in your own settings. This would be the place to add it so that upgrades to the package work seamlessly. Once you make changes to the index settings you need to do [_rollover](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-rollover-index.html) to create a new backing index that uses the new settings.

This is the default index template for the netflow data stream. It already references the `logs-netflow.log@custom` component template (out of the box the component template itself won't exist).

<img width="587" alt="Image" src="https://github.com/user-attachments/assets/620f0627-a72a-4109-9eae-ea0d92360b8d" />

Create the logs-netflow.log@custom component template and add the `index.mode` setting.

<img width="383" alt="Image" src="https://github.com/user-attachments/assets/ddd8122d-bcb5-4779-9cd7-29501b65acaa" />

Preview of the final index mapping that shows index.mode was integrated into the final settings.

<img width="402" alt="Image" src="https://github.com/user-attachments/assets/1e38d5e6-6b2b-47b8-93b1-57e9593cf283" />


[^1]: https://www.elastic.co/guide/en/elasticsearch/reference/current/logs-data-stream.html#how-to-use-logsds
